memlru/memlru.go

```diff
@@ -121,3 +135,21 @@ func (m *MemLRU) Delete(ctx context.Context, key string) error {
 	_ = present
 	return nil
 }
+
+func (m *MemLRU) setKeyValue(ctx context.Context, key string, value []byte) error {
+	if len(key) > cachestore.MaxKeyLength {
+		return cachestore.ErrKeyLengthTooLong
+	}
+	if len(key) == 0 {
+		return cachestore.ErrInvalidKey
+	}
+	m.backend.Add(key, value)
+	return nil
+}
+
+func (m *MemLRU) removeExpiredKeys() {
+	expiredKeys := m.expirationQueue.Expired()
```

Comment:
nice PR, I haven't had a chance to go through everything in detail, but the one thing I would do at the top of this method is check the lastExpiryCheck .. and only check for expired keys every like 5 seconds.. this way if the memlru is being blasted with gets etc.. it will only try to remove keys once every 5 seconds

Reply:
Good idea, I added a check in removeExpiredKeys so that the actual purge of old keys happens after more than `lastExpiryCheckInterval`.